Sources/Logging/Logging.swift

```diff
@@ -39,20 +39,44 @@ import WASILibc
 /// logger.info("Hello World!")
 /// ```
 public struct Logger {
-    @usableFromInline
-    var handler: LogHandler
+    /// A private property to hold the boxed `LogHandler`.
+    private var _handler: Storage
 
     /// An identifier of the creator of this `Logger`.
     public let label: String
+    
+    /// A computed property to access the `LogHandler`.
+    public var handler: LogHandler {
+        get {
+            return _handler.handler
+        }
+        set {
+            if !isKnownUniquelyReferenced(&self._handler) {
+                self._handler = Storage(label: self.label, handler: newValue)
+            } else {
+                self._handler.handler = newValue
+            }
+        }
+    }
 
     /// The metadata provider this logger was created with.
     public var metadataProvider: Logger.MetadataProvider? {
-        return self.handler.metadataProvider
+        return handler.metadataProvider
     }
 
     internal init(label: String, _ handler: LogHandler) {
         self.label = label
-        self.handler = handler
+        self._handler = Storage(label: label, handler: handler)
+    }
+
+    public class Storage: @unchecked Sendable {
+        var label: String
+        var handler: LogHandler
```

Comment:
Can we add `@usableFromInline` to both of these properties and please copy the comment over from the `Logger` properties.